app/assets/javascripts/discourse/app/templates/components/edit-category-topic-template.hbs

```diff
@@ -1,4 +1,31 @@
-<DEditor
-  @value={{this.category.topic_template}}
-  @showLink={{this.showInsertLinkButton}}
-/>
\ No newline at end of file
+{{#if this.siteSettings.experimental_form_templates}}
+  <div class="control-group">
+    <ComboBox
+      @class="select-category-template"
+      @content={{this.templates}}
+      @value={{this.selectedTemplate}}
+      @valueProperty="name"
+      @onChange={{this.changeTemplate}}
+    />
+  </div>
+
+  {{#if (eq this.selectedTemplate this.freeformTemplate.name)}}
+    <DEditor
+      @value={{this.category.topic_template}}
+      @showLink={{this.showInsertLinkButton}}
+    />
+  {{else}}
+    <pre>
+      <code class="yaml hljs language-yaml">{{this.templateContent}}</code>
+    </pre>
+
+    {{! TODO(@keegan): Fix <HiglightedCode> only rendering once }}
+    {{!-- <HighlightedCode @lang="yaml" @code={{this.templateContent}} /> --}}
```

Comment:
Need to find out a fix to successfully show YAML highlighting over code.

When I tried using the `<HighlightedCode />` component, it shows highlighting but the `@code` doesn't update when the `this.templateContent` value changes based on the `<ComboBox>`. If anybody knows a fix for this, it would be much appreciated, if not, I will revisit this `TODO` later as its not urgent.